source/lac/petsc_vector_base.cc

```diff
@@ -705,12 +774,14 @@ namespace PETScWrappers
   VectorBase &
   VectorBase::operator*=(const PetscScalar a)
   {
-    Assert(!has_ghost_elements(), ExcGhostsPresent());
     AssertIsFinite(a);
 
     const PetscErrorCode ierr = VecScale(vector, a);
     AssertThrow(ierr == 0, ExcPETScError(ierr));
 
+    // update ghost values if needed
+    update_ghost_values();
+
```

Comment:
This falls in the same category: One *can* do that, but it doesn't fit into the model we have.